### PR TITLE
Conditional check if cloud is aws for node name check

### PR DESCRIPTION
**Pull request by geoff-elastisys**
**What this PR does / why we need it**:Checks if cloud is AWS.  Node name is then gotten a different way so that it can be tested

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*:21
fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/ck8s-cluster/blob/master/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

### CODE REVIEW

File: client/cluster.go [hunk 1]
```diff
@@ -411,10 +413,25 @@ func (c *ClusterClient) DrainNode(name string) error {
 func (c *ClusterClient) NodeExists(name string) (bool, error) {
 	c.logger.Info("client_node_exists", zap.String("name", name))
 
-	if err := c.kubectl.NodeExists(name); err != nil {
+	addPrefix := true
+
+	if c.cluster.CloudProvider() == api.AWS {
+		myvar, err := c.Machine(name)
```
Comment (Xartos) [inline client/cluster.go]: This variable should have a more descriptive name

```suggestion
		machine, err := c.Machine(name)
```

File: client/cluster.go [hunk 2]
```diff
@@ -411,10 +413,25 @@ func (c *ClusterClient) DrainNode(name string) error {
...
+		if err != nil {
+			return false, err
+		}
+
+		myPrivateIP := strings.Replace(myvar.PrivateIP, ".", "-", -1)
```
Comment (Xartos) [inline client/cluster.go]: Prefixing with "my" feels a bit odd.
```suggestion
		privateIPString := strings.Replace(myvar.PrivateIP, ".", "-", -1)
```

File: client/cluster.go [hunk 3]
```diff
@@ -411,10 +413,25 @@ func (c *ClusterClient) DrainNode(name string) error {
...
+		myRegion := c.cluster.TFVars().(*aws.AWSTFVars).Region
```
Comment (Xartos) [inline client/cluster.go]: Same here

File: runner/kubectl.go [hunk 4]
```diff
@@ -63,10 +63,13 @@ func (k *Kubectl) Command(args []string) error {
 
 // NodeExists runs `sops exec-file KUBECONFIG 'kubectl get node NAME'` in the
 // background. If the node is not found a NodeNotFoundErr error is returned.
-func (k *Kubectl) NodeExists(name string) error {
+func (k *Kubectl) NodeExists(name string, addPrefix bool) error {
 	k.logger.Debug("kubectl_node_exists")
 
-	cmd := k.command("get", "node", k.fullNodeName(name))
+	if addPrefix == true {
```
Comment (Xartos) [inline runner/kubectl.go]: This can be simplified
```suggestion
	if addPrefix {
```

**Review by Xartos — CHANGES_REQUESTED** [review]
Just some small things

File: client/cluster.go (hunk 3 above)
Comment (geoff-elastisys) [reply client/cluster.go]: done

**Review by Xartos — COMMENTED** [review]
I think it looks good now! Although I would like someone else to approve it as well, since I was part of this implementation.
I'm adding @viktor-f as a reviewer as well since simon isn't working atm.

**Review by Xartos — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]
LGTM

**Pull request merged**